### ISSUE TITLE
bug: price inversion fails on call to `toReversed`

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/dex-explorer#127]
During review of #125, I observed a logged error relating to use of the `toReversed()` function:

```
Failed to generate cache key for https://testnet.plinfra.net/penumbra.core.component.dex.v1.QueryService/CandlestickData
Failed to generate cache key for https://testnet.plinfra.net/penumbra.core.component.dex.v1.QueryService/CandlestickData
 ⨯ TypeError: s.toReversed is not a function
    at em (/app/.next/server/app/api/book/route.js:1:26002)
    at eA (/app/.next/server/app/api/book/route.js:1:27380)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36932
    at async eC.execute (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:27548)
    at async eC.handle (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:38186)
    at async doRender (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1359:42)
    at async cacheEntry.responseCache.get.routeKind (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1581:28)
    at async NextNodeServer.renderToResponseWithComponentsImpl (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1489:28)
    at async NextNodeServer.renderPageComponent (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1913:24)
```

The relevant code is here: https://github.com/penumbra-zone/dex-explorer/blob/2f3633d19b7ea2218e475ed36c95ac520246b169/src/shared/api/server/book/helpers.ts#L121-L131 and blame reports last changed in 22b313e0f0fb410b974942bc8e0c7bd09cc70c19. Not sure whether those two calls to `toReversed` are both necessary; perhaps the problem will go away once we straighten out the db, cf. #126.

**Comment by conorsch** [comment]
This error is still occurring on the testnet deployment, as of today.

**Comment by conorsch** [comment]
Still happening. When I load this route: https://dex-explorer.testnet.plinfra.net/api/book?baseAsset=UM&quoteAsset=GN I see the following error:

```
 ⨯ TypeError: a.toReversed is not a function
    at sT (/app/.next/server/app/api/book/route.js:1:194778)
    at fb (/app/.next/server/app/api/book/route.js:1:494089)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36932
    at async eC.execute (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:27548)
    at async eC.handle (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:38186)
    at async doRender (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1359:42)
    at async cacheEntry.responseCache.get.routeKind (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1581:28)
    at async NextNodeServer.renderToResponseWithComponentsImpl (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1489:28)
    at async NextNodeServer.renderPageComponent (/app/node_modules/.pnpm/next@14.2.9_@babel+core@7.25.2_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:1913:24)
-- [2024-11-19T16:23:30.254Z] 0.967 GET /api/book?baseAsset=UM&quoteAsset=GN
```

**Comment by grod220** [comment]
It looks like there is an unexpected type causes a runtime error 🤔 . It's strange that this only occurs in the live deployment and not locally. My guess is that the request to the `SimulationService` may be failing (?). This is quite tough to tell. Is the next best option to debug adding a ton of console logs and re-deploying?

**Comment by conorsch** [comment]
Confirmed fixed after merge/deploy of #138:

![dex-explorer-reversed-fix](https://github.com/user-attachments/assets/b2628a59-6393-4385-ab4e-21d0bf2e6c9a)
